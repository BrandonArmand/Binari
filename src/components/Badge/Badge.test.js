import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Badge from "./Badge";

describe("Badge", () => {
    it("renders snapshot", () => {
        const { container } = render(<Badge />);
        expect(container).toMatchSnapshot();
    });
    it("renders snapshot with colour info", () => {
        render(<Badge color="info">Info Badge</Badge>);
        expect(screen.getByText("Info Badge")).toBeInTheDocument();
        expect(screen.getByText("Info Badge")).toHaveClass(
            "makeStyles-info-14"
        );
    });
    it("renders text", () => {
        render(<Badge>Default Badge</Badge>);
        expect(screen.getByText("Default Badge")).toBeInTheDocument();
        expect(screen.getByText("Default Badge")).toHaveClass(
            "makeStyles-badge-17 makeStyles-gray-24"
        );
    });
});
