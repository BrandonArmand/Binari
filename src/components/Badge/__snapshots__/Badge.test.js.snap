// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders snapshot 1`] = `
<div>
  <span
    class="makeStyles-badge-1 makeStyles-gray-8"
  />
</div>
`;
